src/lib.rs

```diff
@@ -0,0 +1,235 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use discord_presence::Client;
+use rrplug::prelude::*;
+
+const EMPTY: String = String::new();
+
+static mut WAS_IN_GAME: bool = false;
+static mut RESET_SINGLE_PLAYER_TIMER: bool = true;
+
+struct ActivityData {
+    party: (u32, u32),
+    details: String,
+    state: String,
+    large_image: String,
+    large_text: String,
+    small_image: String,
+    small_text: String,
+    end: Option<u64>,
+    start: Option<u64>,
+}
+
+struct DiscordRpc {
+    gamestate: Option<GameState>,
+    serverinfo: Option<ServerInfo>,
+}
+
+impl Plugin for DiscordRpc {
+    fn new() -> Self {
+        Self {
+            gamestate: None,
+            serverinfo: None,
+        }
+    }
+
+    fn initialize(&mut self, external_plugin_data: ExternalPluginData) {
+        self.gamestate = external_plugin_data.get_game_state_struct();
+        self.serverinfo = external_plugin_data.get_server_info_struct();
+        println!("discord rpc plugin initialized");
+    }
+
+    fn main(&self) {
+        let gamestate = self.gamestate.as_ref().unwrap();
+        let serverinfo = self.serverinfo.as_ref().unwrap();
+
+        let mut drpc = Client::new(941428101429231617);
+
+        let _ =drpc.start();
+        println!("discord rpc initialized");
+
+        loop {
+            if let Err(err) = drpc.clear_activity() {
+                #[cfg(debug_assertions)]
+                println!("coudln't clear activity because of {:?}", err)
+            }
+
+            println!("started a cycle");
```

Comment:
How often do we loop? And where does the print show up? Cause if it ends up in the default northstar logs, we might wanna prefix it with `[DiscordRPC]` and maybe get the formatting similar to spdlog ^^